lib/jobs.js

```diff
@@ -61,8 +61,22 @@ function start(buildConfig, callback) {
  * @param {Function}   callback                   Callback fn(error, result)
  */
 function stop(buildConfig, callback) {
-    return redis.hdel(`${queuePrefix}buildConfigs`, buildConfig.buildId)
-        .then(() => executor.stop(buildConfig))
+    const stopId = buildConfig.buildId;
+    const stopConfig = { buildId: stopId };
+
+    return redis.hget(`${queuePrefix}buildConfigs`, stopId)
+        .then((fullBuildConfig) => {
+            const parsedConfig = JSON.parse(fullBuildConfig);
+
+            if (parsedConfig && parsedConfig.annotations) {
+                stopConfig.annotations = parsedConfig.annotations;
+            }
+
+            return redis.hdel(`${queuePrefix}buildConfigs`, stopId);
+        }, (err) => {
```

Comment:
Could you wrap this callback with `.catch()` for readability?

Reply:
fixed.